### PR TITLE
chore(CDAP-19231): convert complex schema tests to cucumber selenium tests

### PR DESCRIPTION
**Pull request by de-lan**
convert complex schema tests to cucumber selenium tests

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19231](https://cdap.atlassian.net/browse/CDAP-19231)

### CODE REVIEW

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/Admin.java [hunk 1]
```diff
@@ -83,19 +79,23 @@ public void checkForErrorInSavingPreferences() {
...
     );
-    if (!addedKey.equals(key) || !addedValue.equals(value)) {
-      throw new RuntimeException(
-        "Saved key-value pair does not match with user provided key-value pair" +
-          "\nUser provided \"" + key + "\" for key and \"" + value + "\" for value." +
-          "\nSaved values are \"" + addedKey + "\" for key and \"" + addedValue + "\" for value."
-      );
-    }
+
+    Assert.assertEquals(addedKey, key);
```
Comment (de-lan) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/Admin.java:98]: Using JUnit to simplify assertion.

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/ComputeProfile.java [hunk 2]
```diff
@@ -25,128 +25,111 @@
 import io.cdap.e2e.utils.WaitHelper;
```
Comment (de-lan) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/ComputeProfile.java:25]: This one does not have code change, mainly re-formatting, indent

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java [hunk 3]
```diff
@@ -93,6 +95,11 @@ public static WebElement locateElementByCssSelector(String cssSelector) {
       .findElement(By.cssSelector(cssSelector));
   }
 
+  public static WebElement locateElementByTestId(String testId) {
```
Comment (de-lan) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java:98]: Added one method for simplification.

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  @Then("Click on Projection node property button")
+  public void clickOnProjectionNodePropertyButton() {
+    CdfStudioActions.navigateToPluginPropertiesPage(projectionNode.getNodeName());
+  }
+
+  @Then("Verify there is one empty row")
+  public void verifyThereIsOneEmptyRow() {
+    WebElement row0Selector = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow1));
+
```
Comment (sumengwang) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java]: empty line

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+  public void removeRowTwo() {
+    WebElement removeButtonElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow2) + " " + Helper.getCssSelectorByDataTestId(removeButton)
+    );
+    ElementHelper.clickOnElement(removeButtonElement);
+  }
+
+  @Then("Verify row three does not exist")
+  public void verifyRowThreeDoesNotExist() {
+    try {
```
Comment (sumengwang) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java]: I think we should use `isElementExists` from helper

**Review by sumengwang — APPROVED** [review]

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java (hunk 5 above)
Comment (de-lan) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java]: nice catch, updated

**Review by seanfeld — APPROVED** [review]

**Pull request merged**